clojure-mode.el

```diff
@@ -1041,16 +1041,13 @@ nil."
 
 Use `default-directory' if DIR-NAME is nil.
 Return nil if not inside a project."
-  (let ((dir-name (or dir-name default-directory)))
-    (let ((lein-project-dir (locate-dominating-file dir-name "project.clj"))
-          (boot-project-dir (locate-dominating-file dir-name "build.boot")))
-      (when (or lein-project-dir boot-project-dir)
-        (file-truename
-         (cond ((not lein-project-dir) boot-project-dir)
-               ((not boot-project-dir) lein-project-dir)
-               (t (if (file-in-directory-p lein-project-dir boot-project-dir)
-                      lein-project-dir
-                    boot-project-dir))))))))
+  (let* ((dir-name (or dir-name default-directory))
+         (choices (delq nil
+                        (mapcar (lambda (fname)
+                                  (locate-dominating-file dir-name fname))
+                                '("project.clj" "build.boot" "build.gradle")))))
+    (when (< 0 (length choices))
```

Comment:
I'd normally write this as `(> (length choices) 0)` as it reads more naturally to me.